test/organisations_api_test.rb

```diff
@@ -63,8 +63,18 @@ def organisation(slug: "test-department")
 
   describe "fetching a paginated list of organisations" do
     let(:api_client_endpoint) { "#{organisation_api_host}/api/organisations" }
-    let(:page_one_links) { %(<#{api_client_endpoint}?page=2>; rel="next", <#{api_client_endpoint}?page=1>; rel="self") }
-    let(:page_two_links) { %(<#{api_client_endpoint}?page=1>; rel="previous", <#{api_client_endpoint}?page=2>; rel="self") }
+    let(:page_one_links) do
+      Pact.term(
+        /^<https?:\/\/(?:[^\/]+)\/api\/organisations\?page=2>; rel="next", <https?:\/\/(?:[^\/]+)\/api\/organisations\?page=1>; rel="self"$/,
+        %(<#{api_client_endpoint}?page=2>; rel="next", <#{api_client_endpoint}?page=1>; rel="self"),
+      )
+    end
+    let(:page_two_links) do
+      Pact.term(
+        /^<https?:\/\/(?:[^\/]+)\/api\/organisations\?page=1>; rel="previous", <https?:\/\/(?:[^\/]+)\/api\/organisations\?page=2>; rel="self"$/,
+        %(<#{api_client_endpoint}?page=1>; rel="previous", <#{api_client_endpoint}?page=2>; rel="self"),
+      )
+    end
```

Comment:
These look great, thanks for your digging into these. 

I have a minor syntax suggestion to make them a bit easier to read:

```
let(:host_agnostic_endpoint_regex) { %r{https?://(?:[^/]+)/api/organisations} }
let(:api_client_endpoint) { "#{organisation_api_host}/api/organisations" }

let(:page_one_links) do
  Pact.term(
    /\A<#{host_agnostic_endpoint_regex}\?page=2>; rel="next", <#{host_agnostic_endpoint_regex}\?page=1>; rel="self"\z/,
    %(<#{api_client_endpoint}?page=2>; rel="next", <#{api_client_endpoint}?page=1>; rel="self"),
  )
end
```

Do you think it'll be also easier to read if you we use the keyword arguments of generate and matcher as per the examples https://docs.pact.io/getting_started/matching/  ? - that might make it clearer what the two arguments mean (though generate is a rather opaque keyword).

Reply:
Good idea - I've amended as suggested, and used the keywords, which do I think improve readability slightly.